### DISCUSSION TITLE
How to protect developers' privacy and customize certificate organizations? Preventing the leakage of device owner's identity information.

### DISCUSSION THREAD

**Discussion by geekyouth** [FiloSottile/mkcert#613] [Ideas]
![image](https://github.com/user-attachments/assets/03d38f62-7e81-4f7c-9468-ed100ec5bf52)
![image](https://github.com/user-attachments/assets/e54a3b5b-d905-4f7d-bf38-6686b81c6ae9)

**Comment by DoraTheExploder** [comment]
I've discovered that if you use an external tool to generate a Certificate Signing Request (.csr) such as [DigiCert CSR Tool](https://www.digicert.com/easy-csr/openssl.htm) (or do it manually with openssl), you can customize the relevant fields and then pass it as a sort of configuration file to mkcert with `$ mkcert -csr <domain>`.

I have not tested using a CSR for initial Root CA installation, but `$ mkcert --help` implies that it should be possible.
